### PR TITLE
Close #31 Added typings

### LINKED ISSUE #31: Add typings for Typescript users
I love your tree control, but since I'm using Typescript, it is not so easy to properly use it. Of course, you can use a hack like 

```ts
(L.control.layers as any).tree...
```

but proper typings would be better.

### PR DESCRIPTION
**Pull request by erikvullings**
I've just opened an issue, which can be closed if you merge this pull request ;-)

No code changes have been added, just one new typings file and a reference to it in the `package.json` file.

### CODE REVIEW

**Comment by erikvullings** [comment]
Minor addition: I've added an `.npmignore` file, so when you publish to npm, the examples and test folder are not pushed, nor the bower.json and travis file. As they are by-products, they are normally not needed, and it will reduce the npm package size. 
Feel free to remove it if you have a different opinion (after all, it's your repo ;-)

File: L.Control.Layers.Tree.d.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      namespace layers {
+        function tree(
+          baseTree: ILayerTree,
+          overlayTree: ILayerTree,
+          options?: Partial<ILayerTreeOptions>
+        ): L.Control;
+      }
+    }
+  }
+}
```
Comment (jjimenezshaw) [inline L.Control.Layers.Tree.d.ts:74]: Sorry, I am not an expert in TypeScript. So I asked a friend of mine.
They suggested this full definition.

(I do not know a better way to give you the full file)
```suggestion
import type L from 'leaflet';

export namespace Control {
  export namespace Layers {
    interface LayerTreeChildren {
      label: string;
      layer: L.Layer;
      selectAllCheckbox?: boolean;
      children?: LayerTreeChildren[];
    }

    interface LayerTree {
      label: string;
      children?: LayerTreeChildren[];
    }

    interface LayerTreeOptions extends L.Control.LayersOptions {
      /**
      * Symbol displayed on a closed node.
      * @default '+'
      */
      closedSymbol?: string;

      /**
      * Symbol displayed on an open node.
      * @default '−' (&minus;)
      */
      openedSymbol?: string;

      /**
      * Symbol between the closed or opened symbol, and the text. Default .
      * @default ' ' (a normal space)
      */
      spaceSymbol?: boolean;

      /**
      * Flag to indicate if the selector (+ or −) is after the text.
      * @default 'false'
      */
      selectorBack?: boolean;

      /**
      * Flag to replace the toggle image (box with the layers image) with the 'name' of the selected base layer.
      * If the name field is not present in the tree for this layer, label is used.
      * See that you can show a different name when control is collapsed than the one that appears in the tree
      * when it is expanded. Your node in the tree can be
      * { label: 'OSM', name: 'OpenStreetMap', layer: layer }.
      * @default 'false'.
      */
      namedToggle?: boolean;

      /**
      * Text for an entry in control that collapses the tree (baselayers or overlays).
      * If empty, no entry is created.
      * @default ''.
      */
      collapseAll?: string;

      /**
      * Text for an entry in control that expands the tree. If empty, no entry is created.
      * @default ''.
      */
      expandAll?: string;

      /**
      * Controls if a label or only the checkbox/radiobutton can toggle layers.
      * If set to both, overlay or base those labels can be clicked on to toggle the layer.
      * @default 'both'.
      */
      labelIsSelector?: 'both' | 'overlay' | 'base' | string;
    }

    class Tree extends L.Control.Layers {
      constructor(baseTree?: LayerTree, overlayTree?: LayerTree, options?: Partial<LayerTreeOptions>);
      addBaseLayer(layer: L.Layer, name: string): never;
      addOverlay(layer: L.Layer, name: string): never;
      removeLayer(layer: L.Layer): never;
      expandTree(isOverlay?: boolean): this;
      collapseTree(isOverlay?: boolean): this;
      expandSelected(isOverlay?: boolean): this;
      options: LayerTreeOptions;
    }
  }
}

export namespace control {
  export namespace layers {
    function tree(
      baseTree?: Control.Layers.LayerTree,
      overlayTree?: Control.Layers.LayerTree,
      options?: Partial<Control.Layers.LayerTreeOptions>
    ): Control.Layers.Tree;
  }
}
```

**Comment by jjimenezshaw** [comment]
Hi @erikvullings any question about it? The changes I suggest are not that many. Just that I had the whole file, and was easier to do this way (replace everything), than cherry pick each one... and it is less error prone. If you replace the file in your computer you will see the actual differences.

File: L.Control.Layers.Tree.d.ts (hunk 1 above)
Comment (erikvullings) [reply L.Control.Layers.Tree.d.ts:74]: @jjimenezshaw Sorry for the delay - missed your first revision. I've tried to copy paste it in our project (https://github.com/reiniernl/nederland-waterland), but alas, that did not work as expected, and I did get Typescript errors in my code. Did you do something special too?

Comment (jjimenezshaw) [reply L.Control.Layers.Tree.d.ts:74]: Hi @erikvullings . Could you check #36 ? (minor things in #35)
Pay attention to `files` in `test-types/tsconfig.json` https://github.com/jjimenezshaw/Leaflet.Control.Layers.Tree/pull/36/files#diff-d2828f04b5fc8ca0298f1358b0f50c12efcd37281d152f941b6da0a4697c3e57R11

**Comment by jjimenezshaw** [comment]
Done in #36

**Pull request closed without merging**